t/34-git.t

```diff
@@ -98,6 +100,51 @@ subtest 'successful clone' => sub {
     is($@, '', 'serialization successful');
 };
 
+subtest 'cloning with caching' => sub {
+    # setup temp dir for cache and configure using it
+    my $git_cache_dir = tempdir('temp-git-caching-XXXXX')->make_path;
+    note "temp dir for cache: $git_cache_dir";
+    $bmwqemu::vars{GIT_CACHE_DIR} = $git_cache_dir->to_string;
+
+    # make up parameters for cloning
+    my ($orga, $repo, $rev) = (qw(os-autoinst os-autoinst-wheel-launcher 742bd0570a5d086be12fecb3b108bff15f4cb202));
+    my $url = Mojo::URL->new("https://github.com/$orga/$repo.git");    # FIXME: make this work offline
```

Comment:
see https://github.com/os-autoinst/os-autoinst/pull/2446/files#diff-e3d0ad755517b7f2d6ae3e420e1f4b0c8d4e81153418d21f2d637fe0ac640da3R58 for this